src/resolve-environment.ts

```diff
@@ -0,0 +1,18 @@
+import { getCodeQL } from "./codeql";
+import { Language } from "./languages";
+import { Logger } from "./logging";
+
+export async function runResolveBuildEnvironment(
+  cmd: string,
+  logger: Logger,
+  workingDir: string | undefined,
+  language: Language
+) {
+  logger.startGroup(
+    `Attempting to resolve build environment for ${language} in ${workingDir}`
+  );
+  const codeQL = await getCodeQL(cmd);
+  const result = await codeQL.resolveBuildEnvironment(workingDir, language);
```

Comment:
Nit: we already have some inconsistency in the repo, but `codeql` is more common.

```suggestion
  const codeql = await getCodeQL(cmd);
  const result = await codeql.resolveBuildEnvironment(workingDir, language);
```